src/main/java/org/folio/circulation/domain/notice/schedule/LoanScheduledNoticeHandler.java

```diff
@@ -250,7 +251,31 @@ public static LoanScheduledNoticeHandler using(Clients clients, DateTime systemT
   }
 
   public boolean noticeIsNotRelevant(ScheduledNotice notice, Loan loan) {
-    return loan.isClosed() || beforeDueDateNoticeIsNotRelevant(notice, loan);
+    if (loan.isClosed()) {
+      return true;
+    }
+    TriggeringEvent triggeringEvent = notice.getTriggeringEvent();
+    if (triggeringEvent == DUE_DATE) {
+      return dueDateNoticeIsNotRelevant(notice, loan);
+    } else if (triggeringEvent == AGED_TO_LOST) {
+      return agedToLostNoticeIsNotRelevant(loan);
+    }
+    log.error("Unexpected triggering event {}", triggeringEvent.getRepresentation());
+    throw new UnsupportedOperationException("Unexpected triggering event: " +
+      triggeringEvent.getRepresentation());
+  }
+
+  private boolean dueDateNoticeIsNotRelevant(ScheduledNotice notice, Loan loan) {
+    return beforeDueDateNoticeIsNotRelevant(notice, loan)
+      || loan.getItem().isDeclaredLost()
+      || loan.getItem().getStatus() == ItemStatus.AGED_TO_LOST
+      || loan.isRenewed()
+      || loan.getItem().isClaimedReturned()
+      || (loan.hasDueDateChanged() && loan.getDueDate().isAfter(systemTime));
```

Comment:
Consider removing this condition, it's already covered in `ChangeDueDateResource`